Core/include/Acts/Utilities/Result.hpp

```diff
@@ -172,6 +175,144 @@ class Result {
     return std::move(std::get<T>(m_var));
   }
 
+  /// Retrieves the valid value from the result object, or returns a default
+  /// value if no valid value exists.
+  ///
+  /// @param[in] v The default value to use if no valid value exists.
+  /// @note This is the lvalue version.
+  /// @note This function always returns by value.
+  /// @return Either the valid value, or the given substitute.
+  template <typename U>
+  T value_or(U&& v) const&
```

Comment:
this will copy `value()` if `ok()`?

Reply:
Yes, I think that makes sense because it's the lvalue version; do we cannot move the value out.

Reply:
but in principle `value()` could allow you to return a `const &` right? but I guess the supplied default makes this impossible?

Reply:
Would an overload taking `U` as (const) lvalue ref and returning everything as (const) lvalue ref make sense?

Reply:
I think I managed to make this work with some conditional typing. Take a look!